From 3b6bc916ff377e41fa2bc32344302cc164a19c56 Mon Sep 17 00:00:00 2001
From: minaripenguin <minaripenguin@users.noreply.github.com>
Date: Fri, 10 Mar 2023 20:18:18 +0800
Subject: [PATCH] ConfigurationController: Prevent app crash caused by
 null configuration

fixes:
* https://bin.kv2.dev/~640b1b26e599576ea02bdd81

test: open app in bug, no crashes after patch

Signed-off-by: minaripenguin <minaripenguin@users.noreply.github.com>
Signed-off-by: Anushek Prasal <anushekprasal@gmail.com>
Signed-off-by: Pranav Vashi <neobuddy89@gmail.com>
---
 core/java/android/app/ConfigurationController.java | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/core/java/android/app/ConfigurationController.java b/core/java/android/app/ConfigurationController.java
index 872b3075c534..a6f227c0465f 100644
--- a/core/java/android/app/ConfigurationController.java
+++ b/core/java/android/app/ConfigurationController.java
@@ -267,6 +267,9 @@ class ConfigurationController {
      * original LocaleList.
      */
     void updateLocaleListFromAppContext(@NonNull Context context) {
+        if (context.getResources() == null || context.getResources().getConfiguration() == null ||
+                mResourcesManager == null || mResourcesManager.getConfiguration() == null)
+            return;
         final Locale bestLocale = context.getResources().getConfiguration().getLocales().get(0);
         final LocaleList newLocaleList = mResourcesManager.getConfiguration().getLocales();
         final int newLocaleListSize = newLocaleList.size();
-- 
2.43.0

